<a href="https://colab.research.google.com/github/TheMaze45/Pandas/blob/main/Recap_Data_Exploration.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Recap on Pandas DataFrames

DataFrames in Pandas are mutable two-dimensional structures of data with labeled axes where:



*   each row represents a different observation
*   each column represents a different variable



Before we can create a DataFrame, we have to import the Pandas module into 
our coding enviroment.

In [2]:
# We can do that with
import pandas as pd # you can name pandas anything you want, but importing pandas as pd is the standard !

# 1. Import a CSV-file (comma seperated values) into the DataFrame


When working with data, most of time you don't create it by yourself, your main job as a Data-Analyst/Scientist is to get data and work with it.

We call this process the importing(or "reading") of data, sometimes from a CSV-file or from a Database.

With Pandas it is easy to import data and convert it into a DataFrame.

Here we take for example a CSV-file that we worked with before.

##Important:

When importing a file from your Google Drive, make sure that the access permission for the file is set to "anyone with the link can read/view the file",otherwise you will have trouble importing the data.

In [3]:
# url is just the link path to your file
url = "https://drive.google.com/file/d/1FYhN_2AzTBFuWcfHaRuKcuCE6CWXsWtG/view?usp=sharing" # orderlines.csv
path = "https://drive.google.com/uc?export=download&id="+url.split("/")[-2]
# This here creates the dataframe
df = pd.read_csv(path)

# Take a quick glance at the dataframe to see if the import worked.
# Don't forget to run the import pandas as pd first !
df.head()

id  id_order  product_id  product_quantity      sku unit_price  \
0  1119109    299539           0                 1  OTT0133      18.99   
1  1119110    299540           0                 1  LGE0043     399.00   
2  1119111    299541           0                 1  PAR0071     474.05   
3  1119112    299542           0                 1  WDT0315      68.39   
4  1119113    299543           0                 1  JBL0104      23.74   

                  date  
0  2017-01-01 00:07:19  
1  2017-01-01 00:19:45  
2  2017-01-01 00:20:57  
3  2017-01-01 00:51:40  
4  2017-01-01 01:06:38

As we can the, the table looks good, we have valid data for every column and row.

#2. DataFrame dimensions

If you don't want or can't count the number of rows and columns by hand on your DataFrame, don't worry, there is a nice little helper



```
df.shape
```
With this attribute we can a quick overview of the size of our table.


In [4]:
df.shape

(293983, 7)

It returns a [tuple](https://www.w3schools.com/python/python_tuples.asp) (data type with multiple items stored in a single var).

The first element of our tuple is the number of rows in our DataFrame, which is 293983 rows, the second element is the amount of columns, for our example it is 7.

If we want to present this kind of information in a nice way, we could extract the values and put the information into a print statement.

In [5]:
nrows = df.shape[0]
ncols = df.shape[1]
print(f'The number of rows in our dataframe is{nrows} and we have a total number of {ncols} columns.')

The number of rows in our dataframe is293983 and we have a total number of 7 columns.


The [DataFrame.size](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.size.html) returns the total number of values that a DataFrame has.

In our case it would be the number of rows (293983) * by the number of columns (7). So a total of 2057881 values.

In [6]:
df.size

2057881

We can verify this with a boolean statement.


```
df.shape[0] * df.shape[1] == df.size
```
If the numbers match, it should return True.


In [7]:
df.shape[0] * df.shape[1] == df.size

True